### PR TITLE
chore: Add NPM duplicate dependency check to canaries

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change adds a script for detecting duplicate Amplify dependencies in NPM and integrates it with our web canaries in CircleCI. Duplicate detection with Yarn will still be handled via our build-time check.

Script sourced from trouble-shooting guide: https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js/#check-for-duplicate-versions

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/duplicates-npm.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# This script detects duplicated Amplify dependencies in the dependency graph (with NPM)
+duplicatedDependencies=$(
```
Comment (stocaaro) [inline .circleci/duplicates-npm.sh:4]: Why are we copying this from the `scripts` location? Code duplication within the package seems like a smell.

**Review by stocaaro — APPROVED** [review]
The duplication seems like a bit more than a nit, but understanding there's probably a reason, I am going to give this an approval.

File: .circleci/config.yml [hunk 2]
```diff
@@ -2512,7 +2512,10 @@ orbs:
               no_output_timeout: 2m
           - run-with-retry:
               label: Install AmplifyJS
-              command: npm i -S aws-amplify && npm i -g wait-on serve
+              command: |
+                npm i -S aws-amplify
+                npm i -g wait-on serve
+                ~/amplify-js/.circleci/duplicates-npm.sh
```
[the remark this answers is not in the document]
Comment (jimblanc) [reply .circleci/config.yml:2518]: This is required for CircleCI to be able to access the script (we do this with some other shell scripts in the `.circleci` folder, e.g. `retry-yarn-script.sh`). The duplication is largely for this reason as well. We could move our build-time script into the `.circleci` folder and adapt it to try to be "smarter" and take a parameter to selectively use NPM or Yarn, but this felt like overkill to me. And our build time script using something in the .circleci folder would also be just as strange as the reverse.

**Review by cshfang — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (stocaaro) [reply .circleci/config.yml:2518]: Can we move the logic into a script, capture the execution in a scripts entry in package.json and then call it in both places to isolate the shared code?

**Pull request merged**